lib/base-compiler.js

```diff
@@ -766,15 +769,50 @@ export class BaseCompiler {
             addOpts.push('-fdump-ipa-all' + flags);
         }
 
-        const newOptions = options.concat(addOpts);
+        // for rustc with GCC backend, wrap the regular options in the misleading 'llvm-args' option.
+        if (isRustcCgGcc) {
+            addOpts = [ '-C' , 'llvm-args=' + addOpts.join(' ')];
+        }
+
+        return options.concat(addOpts);
+    }
+
+    async generateGccDump(inputFilename, options, gccDumpOptions) {
+        let compiler_is_rustc_cg_gcc = (this.compiler.compilerType === 'rustc-cg-gcc');
```

Comment:
We could consider making this an overridable thing in the base class instead of making this special case here. I'm not too worried here, and the ship has sailed already but trying to push per-compiler special casing into the per-compiler classes is best where we can.
Failing that, can this be `const`?

Reply:
Sure, I can try that. I'm really not very fluent in JS so I'm trying to mimic other code or do very simple (sometimes ugly) things, but I am not against doing something cleaner. I'll give it a try and add it here !

Reply:
Got it - if you're not comfortable don't fret too much -- I can take this on if it gets out of hand.